### PR TITLE
feat(core): add runtime flag to disable dummy connector

### PR DESCRIPTION
**Pull request by apoorvdixit88**
## Type of Change

- [ ] Bugfix
- [x] New feature
- [x] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description
Added flag to disable dummy connector in prod.

### Additional Changes

- [ ] This PR modifies the API contract
- [ ] This PR modifies the database schema
- [x] This PR modifies application configuration/environment variables

## How did you test it?
For is_enabled = false;
image

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [x] I reviewed the submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

**Review by ThisIsMani — DISMISSED** [review]

File: crates/router/src/core/admin.rs [hunk 1]
```diff
@@ -439,12 +439,20 @@ fn validate_certificate_in_mca_metadata(
...
+    state: &AppState,
     req: api::MerchantConnectorCreate,
     merchant_id: &String,
 ) -> RouterResponse<api_models::admin::MerchantConnectorResponse> {
-    let key_store = store
-        .get_merchant_key_store_by_merchant_id(merchant_id, &store.get_master_key().to_vec().into())
+    #[cfg(feature = "dummy_connector")]
+    if let Err(error) = validate_dummy_connector_enabled(state, &req.connector_name).await {
+        return Err(error.into());
+    }
```
Comment (NishantJoshi00) [inline crates/router/src/core/admin.rs:449]:
```suggestion
    validate_dummy_connector_enabled(state, &req.connector_name).await?;
```

File: crates/router/src/core/admin.rs [hunk 2]
```diff
@@ -1261,3 +1274,28 @@ pub(crate) fn validate_auth_type(
...
+        match connector_name {
+            api_enums::Connector::DummyConnector1
+            | api_enums::Connector::DummyConnector2
+            | api_enums::Connector::DummyConnector3
+            | api_enums::Connector::DummyConnector4
+            | api_enums::Connector::DummyConnector5
+            | api_enums::Connector::DummyConnector6
+            | api_enums::Connector::DummyConnector7 => {
+                return Err(errors::ApiErrorResponse::InvalidRequestData {
+                    message: format!("{} is not enabled", connector_name),
```
Comment (NishantJoshi00) [inline crates/router/src/core/admin.rs:1294]: Can we change the message here. This message provides significant information about the feature being present in one case and in other. Instead when disabled it shouldn't reveal the existence of this behaviour.

Something like
`invalid connector name`?

File: crates/router/src/core/admin.rs [hunk 3]
```diff
@@ -1261,3 +1274,28 @@ pub(crate) fn validate_auth_type(
...
     }
 }
+
+#[cfg(feature = "dummy_connector")]
+async fn validate_dummy_connector_enabled(
+    state: &AppState,
+    connector_name: &api_enums::Connector,
+) -> Result<(), errors::ApiErrorResponse> {
+    #[cfg(feature = "dummy_connector")]
+    if !state.conf.dummy_connector.is_enabled {
```
Comment (NishantJoshi00) [inline crates/router/src/core/admin.rs:1284]: avoid using return, and here we don't need match. consider using 

`matches!`

**Review by NishantJoshi00 — CHANGES_REQUESTED** [review]

**Review by NishantJoshi00 — DISMISSED** [review]
LGTM

**Review by ThisIsMani — DISMISSED** [review]

File: crates/router/src/configs/settings.rs [hunk 4]
```diff
@@ -159,6 +159,7 @@ where
 #[cfg(feature = "dummy_connector")]
 #[derive(Debug, Deserialize, Clone, Default)]
 pub struct DummyConnector {
+    pub is_enabled: bool,
```
Comment (SanchithHegde) [inline crates/router/src/configs/settings.rs:162]: We could have named this as `enabled` to keep it consistent with the rest of our application config values (take console or file logging configuration for example).

**Review by SanchithHegde — COMMENTED** [review]
Other than that, looks good to me!

**Review by ThisIsMani — APPROVED** [review]

**Review by SanchithHegde — APPROVED** [review]

**Pull request merged**